components/cloudtables/actions/add-single-row/add-single-row.mjs

```diff
@@ -0,0 +1,62 @@
+import cloudtables from "./cloudtables.app.mjs";
+
+export default {
+  key: "cloudtables-add-single-row",
+  name: "Add Single Row",
+  description: "Add a single row of data into CloudTable data set",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    cloudtables,
+    datasetID: {
+      propDefinition: [
+        cloudtables,
+        "datasetID",
+      ],
+      description: "",
+      withLabel: true,
+      reloadProps: true,
+    },
+  },
+  async additionalProps() {
+    const datasetID = this.datasetID?.value || this.datasetID;
+    const dataSetSchema = await this.cloudtables.getDataSetSchema(datasetID);
+    const { datapoints } = dataSetSchema;
+
+    const props = {};
+    for (const datapoint of datapoints) {
+      // the column type is not available in cloudtables API
+      props[`col_${datapoint.id}`] = {
+        type: "string",
+        label: datapoint.name,
+        optional: true,
+      };
+    }
+
+    return props;
+  },
+  async run({ $ }) {
+    const datasetID = this.datasetID?.value || this.datasetID;
+    const dataSetSchema = await this.cloudtables.getDataSetSchema(datasetID);
+    const { datapoints } = dataSetSchema;
+
+    if (datapoints.length === 0) {
+      $.export("$summary", `No data points available at [${datasetID}].`);
```

Comment:
Should this throw an error instead of a success message? I don't know the Cloudtables API - does the user need to take action in this case?